python-client/tests/communications/test_dto_parsing.py

```diff
@@ -0,0 +1,376 @@
+TEST_FUNC_ARGUMENTS_NAME2 = "arg1"
+TEST_FUNC_ARGUMENT_FLOAT = 114.514
+TEST_FUNC_ARGUMENT_INT = 114
+TEST_FUNC_ARGUMENT_STR = "giskard"
+TEST_FUNC_ARGUMENT_BOOL = False
+TEST_FUNC_ARGUMENT_SLICING_FUNCTION_ID = "114"
+TEST_FUNC_ARGUMENT_TRANSFORMATION_FUNCTION_ID = "514"
+
+
+@pytest.mark.parametrize(
+    "func_argument_dict",
+    [
+        {},
+        {"none": True},  # Missing name
+        {"name": TEST_FUNC_ARGUMENT_NAME},  # Missing none
+        {"name": TEST_FUNC_ARGUMENT_NAME, "none": True},  # None argument
+        # {"name": TEST_FUNC_ARGUMENT_NAME, "none": False},   # Not-None argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "model": {"project_key": TEST_PROJECT_KEY, "id": TEST_MODEL_ID},
+        },  # Model argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "dataset": {"project_key": TEST_PROJECT_KEY, "id": TEST_DATASET_ID, "sample": False},
+        },  # Dataset argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "float": TEST_FUNC_ARGUMENT_FLOAT,
+        },  # float argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "int": TEST_FUNC_ARGUMENT_INT,
+        },  # int argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "str": TEST_FUNC_ARGUMENT_STR,
+        },  # str argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "bool": TEST_FUNC_ARGUMENT_BOOL,
+        },  # bool argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "slicingFunction": {"project_key": TEST_PROJECT_KEY, "id": TEST_FUNC_ARGUMENT_SLICING_FUNCTION_ID},
+        },  # slicing function argument
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "transformationFunction": {
+                "project_key": TEST_PROJECT_KEY,
+                "id": TEST_FUNC_ARGUMENT_TRANSFORMATION_FUNCTION_ID,
+            },
+        },  # transformation function argument
+        # kwargs are not used yet
+        {
+            "name": TEST_FUNC_ARGUMENT_NAME,
+            "none": False,
+            "args": [
+                {
+                    "name": TEST_FUNC_ARGUMENTS_NAME1,
+                    "none": False,
+                    "int": TEST_FUNC_ARGUMENT_INT,
+                },  # int argument
+                {
+                    "name": TEST_FUNC_ARGUMENTS_NAME2,
+                    "none": False,
+                    "str": TEST_FUNC_ARGUMENT_STR,
+                },  # float argument
+            ],
+        },  # args function argument
+    ],
+)
+def test_func_argument(func_argument_dict):
+    if "name" not in func_argument_dict.keys() or "none" not in func_argument_dict.keys():
+        with pytest.raises(ValidationError):
```

Comment:
TBH since there's no common validation steps in the test but just a bug if-else, I think it's cleaner to split it into a set of smaller tests. Using parameterized test is already better than having a loop in one test as it was before, but having tests small and focused it equally important (for instance to make it easier to run just one given test)

Reply:
TODO